### PR TITLE
SAA-2036: Add an endpoint to retrieve a count of appointments after a start date grouped by category

### PR DESCRIPTION
**Pull request by dafhoustonmoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/AppointmentCountSummary.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  description =
+  """
+  A count summary of appointments based on a category
+  """,
+)
+data class AppointmentCountSummary(
+
+  @Schema(
+    description = "The appointment category",
+    example = "3",
```
Comment (munslowl) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/AppointmentCountSummary.kt:15]: Remove example?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/appointment/MigrateAppointmentController.kt [hunk 2]
```diff
@@ -153,4 +157,42 @@ class MigrateAppointmentController(
...
+    prisonCode: String,
+    @RequestParam(value = "startDate", required = true)
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
+    @Parameter(description = "Inclusive start date of migrated appointments. Must be today or in the future")
+    startDate: LocalDate,
+    @RequestParam(value = "categoryCodes", required = true)
+    @Parameter(description = "A list of category codes to retrieve summaries appointments")
+    categoryCodes: List<String>,
+  ): List<AppointmentCountSummary> {
+    require(startDate.isAfter(LocalDate.now().minusDays(1))) {
```
Comment (munslowl) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/appointment/MigrateAppointmentController.kt:193]:
```suggestion
    require(startDate >= LocalDate.now()) {
```

File: src/test/resources/test_data/seed-migrated-appointments-summary.sql
Comment (munslowl) [inline src/test/resources/test_data/seed-migrated-appointments-summary.sql:1]: Is it worth some data where prison is not RSI?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/AppointmentCountSummary.kt (hunk 1 above)
Comment (dafhoustonmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/AppointmentCountSummary.kt:15]: will do

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/appointment/MigrateAppointmentController.kt (hunk 2 above)
Comment (dafhoustonmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/appointment/MigrateAppointmentController.kt:193]: neater. I'll change other one in the controller as well.

File: src/test/resources/test_data/seed-migrated-appointments-summary.sql
Comment (dafhoustonmoj) [reply src/test/resources/test_data/seed-migrated-appointments-summary.sql:1]: ok will add a couple and a another test

Comment (dafhoustonmoj) [reply src/test/resources/test_data/seed-migrated-appointments-summary.sql:1]: done

**Review by munslowl — APPROVED** [review]

**Pull request merged**